### PR TITLE
PR13 - Remove AlicaEngine::this from ExpressionHandler

### PR DESCRIPTION
**Pull request by triccyx**
PR13 - Remove AlicaEngine::this from ExpressionHandler and subclasses

### CODE REVIEW

File: alica_engine/include/engine/PlanInterface.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#pragma once
```
Comment (Dale-Koenig) [inline alica_engine/include/engine/PlanInterface.h]: Seems this file is not used in this pr, so push it back to be reviewed in the pr where it is used

File: alica_engine/src/engine/AlicaContext.cpp [hunk 2]
```diff
@@ -101,7 +101,7 @@ void AlicaContext::stepEngine()
         if (std::chrono::system_clock::now() > start + timeout) {
             throw std::runtime_error("Got stuck trying to step engine");
         }
-    } while (!_engine->getPlanBase().isWaiting());
+    } while (!_engine->editPlanBase().isWaiting()); //<<<-----edit instedad of get
```
Comment (Dale-Koenig) [inline alica_engine/src/engine/AlicaContext.cpp]: Remove comment

File: alica_engine/src/engine/planselector/TaskAssignmentProblem.cpp [hunk 3]
```diff
@@ -29,6 +29,8 @@ TaskAssignmentProblem::~TaskAssignmentProblem() {}
 
 /**
  * Constructor of a new TaskAssignment
+ * @brief
```
Comment (Dale-Koenig) [inline alica_engine/src/engine/planselector/TaskAssignmentProblem.cpp]: Undo

Comment (triccyx) [reply alica_engine/src/engine/planselector/TaskAssignmentProblem.cpp]: ok

File: alica_engine/src/engine/AlicaContext.cpp (hunk 2 above)
Comment (triccyx) [reply alica_engine/src/engine/AlicaContext.cpp]: ok

File: alica_engine/include/engine/PlanInterface.h (hunk 1 above)
Comment (triccyx) [reply alica_engine/include/engine/PlanInterface.h]: No more used so removed

**Review by Dale-Koenig — APPROVED** [review]

**Review by veerajsk — APPROVED** [review]

**Pull request merged**